### PR TITLE
Support for DistroScale ads in AMP

### LINKED ISSUE #6293: DistroScale ad support on AMP
Hi,

We would like to add DistroScale into your ad network list. We're ready to open a PR, shall we proceed?

Best,
Jim

### PR DESCRIPTION
**Pull request by jtsai783**
Closes #6293

### CODE REVIEW

**Comment by duancg** [comment]
I signed it!

**Comment by jtsai783** [comment]
I signed it!

File: ads/distroscale.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+```
+
+## Configuration
+
+For semantics of configuration, please [contact DistroScale](http://www.distroscale.com).
+
+__Required:__
+
+- `data-pid` - Partner ID
+- `data-zid` - Zone ID
```
Comment (zhouyx) [inline ads/distroscale.md]: I see you accept `data-tid` as an optional param, please also update in doc

**Comment by zhouyx** [comment]
@jtsai783 Thanks for the Pull Request. Looks good in general.
One thing: we have a [renderStart API](https://github.com/ampproject/amphtml/blob/master/ads/README.md#available-apis) that helps AMP to provide better ad loading experience. 
Could you take a look and implement it (renderStart + noContentAvailable)?
Let me know if there's any question.

File: ads/distroscale.md (hunk 1 above)
Comment (jtsai783) [inline ads/distroscale.md]: `data-tid` is for debugging purposes only and not intended for use by pubs. we'd still like the convenience but wouldn't want to make it publicized knowledge.

**Comment by jtsai783** [comment]
hi, is there an update on this PR?

File: ads/distroscale.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
```
Comment (zhouyx) [inline ads/distroscale.js]: nit: 2016

**Comment by zhouyx** [comment]
@jtsai783 Thanks for supporting renderStart. Looks good. However when tested with your example, I can't get a valid ad. Always get "Error: timeout waiting for ad response". Could you take a look at it. I suppose `render-start` or `no-content` is not being received correctly.  Thanks

**Comment by jtsai783** [comment]
hey @zhouyx i've updated our end to call the renderStart at the appropriate time, however the "a4a should render a single AMP ad in a friendly iframe" test is failing. would you have a clue as to why that is?

**Comment by zhouyx** [comment]
@jtsai783 Thanks. I am able to see the ad now. That test is a flaky one. I think we are fixing that. don't worry about it.

**Comment by jtsai783** [comment]
hey @zhouyx , are further actions needed on our part?

**Comment by zhouyx** [comment]
@jtsai783 code looks good to me. Can you rebase to fix the build failure. 
Also ping @lannka to take a  look.

**Comment by jtsai783** [comment]
hey @zhouyx does the rebasing look correct to you? how should i deal with the CLAs?

**Comment by zhouyx** [comment]
Hi @jtsai783 This rebase doesn't look correct to me. 
For the CLA, I think it will be fixed when you have the rebase correct.

**Comment by jtsai783** [comment]
@zhouyx hi, i think i fixed the errors when i rebased. would you mind reviewing it again?

**Comment by zhouyx** [comment]
LGTM, ping @lannka

File: ads/distroscale.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function distroscale(global, data) {
+  validateData(data, ['pid', 'zid'], ['tid']);
+  let src = '//c.jsrdn.com/s/cs.js?p=' + encodeURIComponent(data.pid) +
+  '&z=' + encodeURIComponent(data.zid) +
+  '&f=' + encodeURIComponent(global.context.location.href);
+  if (data.tid) {
+    src = src + '&t=' + encodeURIComponent(data.tid);
```
Comment (lannka) [inline ads/distroscale.js]: nit: src += ...

File: ads/distroscale.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import {loadScript, validateData} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function distroscale(global, data) {
+  validateData(data, ['pid', 'zid'], ['tid']);
+  let src = '//c.jsrdn.com/s/cs.js?p=' + encodeURIComponent(data.pid) +
+  '&z=' + encodeURIComponent(data.zid) +
```
Comment (lannka) [inline ads/distroscale.js]: indent 4 more spaces

Comment (jtsai783) [inline ads/distroscale.js]: is 1 more level of tab indent ok?

Comment (zhouyx) [inline ads/distroscale.js]: should indent 4 spaces since it's one line of code that can't fit.

**Comment by zhouyx** [comment]
Thanks @jtsai783! Merging the Pull Request.

**Pull request merged**